streaming_data_types/arrays_wa00.py

```diff
@@ -0,0 +1,144 @@
+from typing import Union, Optional, NamedTuple
+from datetime import datetime, timezone
+from dataclasses import dataclass
+
+import flatbuffers
+import numpy as np
+from streaming_data_types.fbschemas.arrays_wav_00.WaveFormArray import WaveFormArray
+from streaming_data_types.fbschemas.arrays_wav_00.DType import DType
+from streaming_data_types.fbschemas.arrays_wav_00.WaveFormArray import (
+    WaveFormArrayStart,
+    WaveFormArrayAddXData,
+    WaveFormArrayAddYData,
+    WaveFormArrayAddXDataType,
+    WaveFormArrayAddYDataType,
+    WaveFormArrayAddYTimestamp,
+    WaveFormArrayAddXTimestamp,
+    WaveFormArrayAddYUnit,
+    WaveFormArrayAddXUnit,
+    WaveFormArrayAddNumberOfElements,
+    WaveFormArrayEnd,
+)
+from streaming_data_types.utils import check_schema_identifier
+
+FILE_IDENTIFIER = b"wa00"
+
+
+def serialise_wa00(
+    values_x_array: np.ndarray,
+    values_y_array: np.ndarray,
+    y_timestamp: Optional[datetime] = None,
+    x_timestamp: Optional[datetime] = None,
+    y_unit: Optional[str] = None,
+    x_unit: Optional[str] = None,
+    number_of_elements: Optional[np.uint32] = None
```

Comment:
Instead of accepting this as an argument, you should calculate it from the arrays.

And while you're at it, you might as well check that they have the same length and are both 1-dimensional.